### PR TITLE
resolves #373

### LINKED ISSUE #373: Crash on Andoird 7.0 after open new Page with map.
Hi. 
I had a crash on android when I open a new straintsu with the map.

Object reference not set to an instance of an object.
```
BaseCameraLogic`1[TNativeMap].Register (Xamarin.Forms.GoogleMaps.Map map, TNativeMap nativeMap)
MapRenderer+<OnElementChanged>d__20.MoveNext ()
ExceptionDispatchInfo.Throw ()
AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_0 (System.Object state)
SyncContext+<>c__DisplayClass2_0.<Post>b__0 ()
Thread+RunnableImplementor.Run ()
IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this)
(wrapper dynamic-method) System.Object:a2f26a92-6781-4bf8-bed7-51cca59648b9 (intptr,intptr)
```

### PR DESCRIPTION
**Pull request by cbagpipe**
#373 
Because of multithreading, Element and Control may be null, so we can't use it in our code. But those properties only contains references to objects we creates. So i decided to save references to our objects in our local fields to be sure we can have those objects when we need.

### CODE REVIEW

**Comment by cbagpipe** [comment]
I didn't properly test it... now pins not displayed on map =(

**Comment by cbagpipe** [comment]
Hmm... Undisplayable pins seems not related to my code modifications, so reopen request.

File: Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/MapRenderer.cs [hunk 1]
```diff
@@ -51,7 +51,19 @@ public MapRenderer() : base()
 
         protected GoogleMap NativeMap { get; private set; }
 
-        protected Map Map => Element;
+        private Map map;
```
Comment (amay077) [inline Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/MapRenderer.cs]: please use ``_`` for private field. (``map`` -> ``_map``)

File: Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/MapRenderer.cs [hunk 2]
```diff
@@ -51,7 +51,19 @@ public MapRenderer() : base()
...
+        protected Map Map
+        {
+            get { return map ?? Element; }
+            set { map = value; }
+        }
+
+        MapView mapView;
```
Comment (amay077) [inline Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/MapRenderer.cs]: please use ``_`` for private field. (``mapView`` -> ``_mapView``)

File: Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/MapRenderer.cs [hunk 3]
```diff
@@ -51,7 +51,19 @@ public MapRenderer() : base()
...
+        public MapView MapView
+        {
+            get { return mapView ?? (MapView)Control; }
+            set { mapView = value; }
```
Comment (amay077) [inline Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/MapRenderer.cs]: Improve accessibility.
``set { mapView = value; }`` to ``private set { mapView = value; }``

File: Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/MapRenderer.cs [hunk 4]
```diff
@@ -298,7 +308,7 @@ private void UpdateHasScrollEnabled(bool? initialScrollGesturesEnabled = null)
         }
 
         private void UpdateHasZoomEnabled(
-            bool? initialZoomControlsEnabled = null,
```
Comment (amay077) [inline Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/MapRenderer.cs]: Unintended changes for this PR. Please revert this row.

File: Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/MapRenderer.cs [hunk 5]
```diff
@@ -323,7 +333,7 @@ private void UpdateIndoorEnabled()
         private void UpdateMapStyle()
         {
             NativeMap.SetMapStyle(Map.MapStyle != null ?
-                new MapStyleOptions(Map.MapStyle.JsonStyle) :
```
Comment (amay077) [inline Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/MapRenderer.cs]: Unintended changes for this PR. Please revert this row.

File: Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/MapRenderer.cs [hunk 6]
```diff
@@ -358,8 +368,8 @@ void SetMapType()
         void SetPadding()
         {
             NativeMap?.SetPadding(
-                (int)(Map.Padding.Left * _scaledDensity), 
-                (int)(Map.Padding.Top * _scaledDensity), 
+                (int)(Map.Padding.Left * _scaledDensity),
+                (int)(Map.Padding.Top * _scaledDensity),
```
Comment (amay077) [inline Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/MapRenderer.cs]: Unintended changes for this PR. Please revert these rows.

File: Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/MapRenderer.cs [hunk 7]
```diff
@@ -381,26 +391,34 @@ public bool OnMyLocationButtonClick()
...
+                     ),
+                 dlat,
+                 dlong
+             );
+
+            }
+            catch (System.Exception ex)
+            {
+                System.Diagnostics.Debug.WriteLine(ex);
+            }
```
Comment (amay077) [inline Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/MapRenderer.cs]: Unintended changes for this PR. Please revert these rows.

File: Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/MapRenderer.cs [hunk 8]
```diff
@@ -447,7 +465,7 @@ protected virtual void OnMarkerDeleted(Pin outerItem, Marker innerItem)
 
         #endregion
 
-        private void Uninitialize(GoogleMap nativeMap, Map map) 
+        private void Uninitialize(GoogleMap nativeMap, Map map)
```
Comment (amay077) [inline Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/MapRenderer.cs]: Unintended changes for this PR. Please revert this row.

**Comment by amay077** [comment]
> I didn't properly test it... now pins not displayed on map =(
> Hmm... Undisplayable pins seems not related to my code modifications, so reopen request.

I think your changes are not related, too. 👍

**Comment by amay077** [comment]
> I didn't properly test it... now pins not displayed on map =(
> Hmm... Undisplayable pins seems not related to my code modifications, so reopen request.

Oops.., This is my bug. 🙇 

I fixed immediately (commit - https://github.com/amay077/Xamarin.Forms.GoogleMaps/commit/1c45cf08d28a46294d305783c5532a68d83cda6c) and released [v2.3.0-beta6](https://www.nuget.org/packages/Xamarin.Forms.GoogleMaps/2.3.0-beta6)

**Comment by amay077** [comment]
Sorry I reject it because no longer response. 🙇

**Pull request closed without merging**